### PR TITLE
Improve scanner coverage for encoded payloads and cross-transport DLP

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## Summary

- Response scanner now decodes base64/hex segments embedded in mixed text, not just fully-encoded content
- MCP input scanning tries pairwise field concatenation for split-value DLP detection
- JSON unicode escape sequences (`\uXXXX`) decoded before DLP scanning on all input paths
- Per-field DLP scanning no longer stops at first match
- Policy matching handles deeper shell nesting and wider token windows
- Extended confusable pre-normalization for policy matching
- Increased WebSocket cross-message DLP overlap window

## Test plan

Exploit-style regression tests for each change. Full test suite passes with race detector.

## Summary by CodeRabbit

* **Improvements**
  * Detect secrets encoded as JSON \uXXXX escapes and re-scan unescaped text.
  * Two-stage decoding: now scans decoded base64/hex segments inside larger text and filters by printable content before matching.
  * Wider cross-message scanning window (512 → 4096 bytes) to catch secrets split across messages.
  * Enhanced policy normalization for additional Cyrillic confusables and deeper iterative resolution.

* **Bug Fixes**
  * Ensure all extracted fields/pairs are evaluated (no premature loop termination).
  * Early-return on first confirmed detection to reduce unnecessary work.

* **Tests**
  * Added unit and integration tests covering new detection paths, edge cases, and regressions.

### CODE REVIEW

File: internal/mcp/policy/policy.go [hunk 1]
```diff
@@ -104,6 +104,10 @@ var shellQuoteStripper = strings.NewReplacer("$'", "", `$"`, "", "'", "", `"`, "
 var policyPreNormalize = strings.NewReplacer(
 	"\u0443", "u", // Cyrillic у — used as 'u' in curl/sudo/su/run
 	"\u0423", "U", // Cyrillic У (uppercase)
+	"\u0432", "b", // Cyrillic в — used as 'b' in bash/base64
+	"\u0412", "B", // Cyrillic В (uppercase)
+	"\u043D", "n", // Cyrillic н — used as 'n' in node/npm/nc
+	"\u041D", "N", // Cyrillic Н (uppercase)
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/mcp/policy/policy.go:118]: Fixed in 1894fed. Implemented dual-view normalization: the policy matcher now checks both the policy-normalized form (в→b, н→n for bash/node) and the baseline confusable form (в→v, н→h for mv/shred/vi/sh). A match on either view triggers the rule. Added 15 test cases covering all bypass scenarios.

File: internal/scanner/response.go [hunk 2]
```diff
@@ -252,9 +257,84 @@
...
+	}
+	return nil
+}
+
+// extractEncodedRuns finds contiguous runs of base64/hex alphabet characters
+// at least minLen long. Returns the segments without surrounding text.
+func extractEncodedRuns(content string, minLen int) []string {
+	var runs []string
+	start := -1
+	for i := 0; i <= len(content); i++ {
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/scanner/response.go]: Outdated. The off-by-one was fixed in a prior commit.

File: internal/mcp/input.go [hunk 3]
```diff
@@ -956,11 +1001,17 @@ func joinStrings(ss []string) string {
...
 }
 
-// scanSplitSecret checks for secrets split across multiple JSON fields by
-// concatenating values without separators. Keys are excluded (via
-// jsonrpc.ExtractStringsFromJSON, not extract.AllStringsFromJSON) because interleaved
-// keys break DLP regex adjacency. Returns the original result if clean or if
-// concat adds no new information.
+// maxPairwiseSplitFields caps the number of field values considered for
+// pairwise split-secret scanning. O(n^2) pairs, but each DLP scan is fast.
+const maxPairwiseSplitFields = 32
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/mcp/input.go]: Fixed in 6f2feba. Bumped maxPairwiseSplitFields from 32 to 64. Edge sampling now checks first/last 32 fields each.

File: internal/scanner/response.go [hunk 4]
```diff
@@ -252,9 +257,85 @@ func (s *Scanner) matchDecodedResponse(content string) []ResponseMatch {
...
+			base64.StdEncoding, base64.URLEncoding,
+			base64.RawStdEncoding, base64.RawURLEncoding,
+		} {
+			if decoded, err := enc.DecodeString(seg); err == nil && len(decoded) > 0 && isPrintableASCII(decoded) {
+				if matches := s.matchDecodedNormalized(string(decoded)); len(matches) > 0 {
+					return matches
+				}
+			}
+		}
+		if decoded, err := hex.DecodeString(seg); err == nil && len(decoded) > 0 && isPrintableASCII(decoded) {
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/scanner/response.go]: Fixed in 6f2feba. Replaced ASCII-only isPrintableASCII with isPrintableText using unicode.IsPrint + utf8.Valid. Valid non-ASCII Unicode now passes through to the normalizer.

File: internal/scanner/response.go [hunk 5]
```diff
@@ -252,9 +257,85 @@ func (s *Scanner) matchDecodedResponse(content string) []ResponseMatch {
...
+// extractEncodedRuns finds contiguous runs of base64/hex alphabet characters
+// at least minLen long. Returns the segments without surrounding text.
+func extractEncodedRuns(content string, minLen int) []string {
+	var runs []string
+	start := -1
+	for i := 0; i < len(content); i++ {
+		c := content[i]
+		inAlphabet := (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') ||
+			(c >= '0' && c <= '9') || c == '+' || c == '/' ||
+			c == '-' || c == '_' || c == '='
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/scanner/response.go]: Fixed in 6f2feba. Removed = from the encoded run character class. Added attachBase64Padding helper that re-attaches trailing padding after extraction, preventing key=payload collapsing.

**Review by DeathCamel58 — APPROVED** [review]
